### PR TITLE
Ignores any defined but not existing cloud-init paths

### PR DESCRIPTION
**Pull request by davidcassany**
Part of rancher/elemental#614

Signed-off-by: David Cassany <dcassany@suse.com>

### CODE REVIEW

**Comment by davidcassany** [comment]
just to not fail if we ever configure a non existing directory over a RO filesystem (ISO filesystem for instance)

**Comment by davidcassany** [comment]
Note gomega is updated in a separated commit, tests were failing so as it was a small not really noisy version bump I just applied it in this same PR.

**Comment by davidcassany** [comment]
Finally added a default path to cloud-init files when booting from ISO. I did not remember at first there was a `cos.setup` parameter parsed in the cmdline to include an extra cloud-init URI. In addition the changes in this PR also allow setting the cloud-init paths in the config file to remote locations, files (not directories) and whatever yip supports.

**Comment by davidcassany** [comment]
Still have to run a manual test, but the expectation is that ISOs built with this elemental-cli version will load cloud-init files included in an `/iso-config` folder with the ISO filesystem root. That folder could include cloud-init files for the ISO boot itself and it could also include installation hooks. Any `elemental run-stage` command executed from the ISO parses the `/proc/cmdline`  and this includes `cos.setup=/run/initramfs/live/iso-config` so this would be parsed at boot and at install time.

File: pkg/constants/constants.go [hunk 1]
```diff
@@ -118,11 +118,12 @@ const (
...
-	IsoInitrdPath = "/boot/initrd"
-	IsoRootFile   = "rootfs.squashfs"
-	IsoEFIImg     = "uefi.img"
-	ISOLabel      = "COS_LIVE"
+	IsoKernelPath    = "/boot/kernel"
+	IsoInitrdPath    = "/boot/initrd"
+	IsoRootFile      = "rootfs.squashfs"
+	IsoEFIImg        = "uefi.img"
+	ISOLabel         = "COS_LIVE"
+	ISOCloudInitPath = LiveDir + "/iso-config"
```
Comment (frelon) [inline pkg/constants/constants.go:126]: Nit: mixing of ISO and Iso in variable name

**Review by frelon — APPROVED** [review]
LGTM

File: pkg/constants/constants.go (hunk 1 above)
Comment (davidcassany) [reply pkg/constants/constants.go:126]: You are right indeed, probably this is the right time to fix it. Gonna make follow up PR if you don't mind :+1: Thanks for the notice though

**Pull request merged**